src/Hal/Application/Config/File/ConfigFileReaderJson.php

```diff
@@ -72,6 +72,13 @@ public function read(Config $config)
             $config->set('git', $jsonData['plugins']['git']['binary']);
         }
 
+        // retrocompatibility with typo in documentation
+        // see https://github.com/phpmetrics/PhpMetrics/issues/441
+        // file -> report
+        if (isset($jsonData['plugins'], $jsonData['plugins']['junit'], $jsonData['plugins']['junit']["file"])) {
+            $jsonData['plugins']['junit']["report"] = $jsonData['plugins']['junit']["file"];
+        }
+
         if (isset($jsonData['plugins'], $jsonData['plugins']['junit'], $jsonData['plugins']['junit']["report"])) {
```

Comment:
Same remarks as for `'file` key, you can reduce the size of your `isset` and replace double quotes with simple quotes for homogeneity.

Reply:
Hi,

isset throw warning on old PHP versions. We need to check each depth of array to avoid warnings.

Reply:
Tested with 3v4l.org through PHP 4.3.0 to PHP 8.0.3 and no warning shows: https://3v4l.org/MYRHv

This is not a configuration that does not display it, because warnings/notices are displayed in the case we are using the variable: https://3v4l.org/U2rnH

If you can experiment a warning without checking all depth, I'm curious to see how and do some tests about this.